core/Updates/5.0.0-b1.php

```diff
@@ -110,4 +117,39 @@ private function requiresUpdatedLogVisitTableIndex()
 
         return true;
     }
+
+    /**
+     * Add new 'pages before column' to log conversions
+     *
+     * @return DbAlias\AddColumns
+     */
+    private function getPagesBeforeAddColumn()
+    {
+        return $this->migration->db->addColumns('log_conversion', ['pageviews_before' => "SMALLINT UNSIGNED DEFAULT NULL"]);
+    }
+
+    private function populatePagesBefore(): void
+    {
+        // Abort if host is *.matomo.cloud
+        $piwikUrl = SettingsPiwik::getPiwikUrl();
+        if (strpos($piwikUrl, '.matomo.cloud') !== false) {
```

Comment:
fyi there could be also `.innocraft.cloud`. And we would need to use `SettingsPiwik::getPiwikInstanceId()` as there could be various custom domains on it and then it wouldn't be detected.